src/components/LayoutTree.tsx

```diff
@@ -358,6 +358,8 @@ const LayoutTree = () => {
       : 0,
   )
 
+  const tableCol1WidthRef = useRef(0)
```

Comment:
Why use a ref? This value should not be preserved across renders.

Reply:
The useMemo hook which is where this ref is being assigned runs after the drag, but we need that during the drag process so the margin can be assigned to the drag-end, so I thought of using ref so I could access the width during the drag.

Reply:
See if you can do this without introducing additional mutable state. If it doesn't seem possible, let me know and I'll do a closer analysis myself. I have a feeling there is a more elegant solution here.